tools/azure-pipelines/build-apache-repo.yml

```diff
@@ -48,3 +58,60 @@ stages:
           e2e_pool_definition:
             vmImage: 'ubuntu-16.04'
           environment: PROFILE="-Dhadoop.version=2.8.3 -Dinclude_hadoop_aws -Dscala-2.11"
+          run_end_to_end: false
+          container: flink-build-jdk8
+  # Special stage for nightly builds:
+  - stage: cron_build
+    displayName: "Cron build"
+    dependsOn: [] # depending on an empty array makes the stages run in parallel
+    condition: or(eq(variables['Build.Reason'], 'Schedule'), eq(variables['MODE'], 'nightly'))
+    jobs:
+      - template: jobs-template.yml
+        parameters:
+          stage_name: cron_build_hadoop241
+          test_pool_definition:
+            name: Default
+          e2e_pool_definition:
+            vmImage: 'ubuntu-16.04'
+          environment: PROFILE="-Dhadoop.version=2.4.1 -Pskip-hive-tests"
+          run_end_to_end: true
+          container: flink-build-jdk8
+      - template: jobs-template.yml
+        parameters:
+          stage_name: cron_build_scala2_12
+          test_pool_definition:
+            name: Default
+          e2e_pool_definition:
+            vmImage: 'ubuntu-16.04'
+          environment: PROFILE="-Dhadoop.version=2.8.3 -Dinclude_hadoop_aws -Dscala-2.12 -Phive-1.2.1"
+          run_end_to_end: true
+          container: flink-build-jdk8
+      - template: jobs-template.yml
+        parameters:
+          stage_name: cron_build_jdk11
+          test_pool_definition:
+            name: Default
+          e2e_pool_definition:
+            vmImage: 'ubuntu-16.04'
+          environment: PROFILE="-Dhadoop.version=2.8.3 -Dinclude_hadoop_aws -Dscala-2.11 -Djdk11"
+          run_end_to_end: true
+          container: flink-build-jdk11
+      - template: jobs-template.yml
+        parameters:
+          stage_name: cron_build_hadoopfree
+          test_pool_definition:
+            name: Default
+          e2e_pool_definition:
+            vmImage: 'ubuntu-16.04'
+          environment: PROFILE=""
+          run_end_to_end: true
+          container: flink-build-jdk8
+      - job: docs_404_check # run on a MSFT provided machine
```

Comment:
where are you caching the gem artifacts?

Reply:
I'm not caching them at all. But this job passes in 5 minutes, so I'm not concerned.

Reply:
it's not a matter of build times but of stability and resource conservation.